src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -33409,6 +33408,299 @@ static void G(object? x)
                 Diagnostic(ErrorCode.WRN_NullReferenceArgument, "x").WithArguments("o", "void C.F(object o)").WithLocation(18, 19));
         }
 
+        [Fact]
+        public void IsDeclarationPattern_01()
+        {
+            var source =
+@"class Program
+{
+    static void F1(object x1)
+    {
+        if (x1 is string y1)
+        {
+            x1.ToString();
+            y1.ToString();
```

Comment:
>y1.ToString(); [](start = 12, length = 14)

Is ```y1``` not nullable or oblivious here? Please add ```y1?.ToString();``` above this line. I would expect to get a diagnostics about unnecessary null check.